### PR TITLE
Create unspecified channel object for unknown channel types

### PR DESCRIPTION
**Pull request by AntiTcb**
Partial resolution of #810. Yay Discord for introducing breaking changes out of left field.

### CODE REVIEW

File: src/Discord.Net.WebSocket/Entities/Channels/SocketGuildChannel.cs [hunk 1]
```diff
@@ -35,7 +35,8 @@ internal static SocketGuildChannel Create(SocketGuild guild, ClientState state,
                 case ChannelType.Voice:
                     return SocketVoiceChannel.Create(guild, state, model);
                 default:
-                    throw new InvalidOperationException("Unknown guild channel type");
+                    // TODO: Proper implementation for channel categories
+                    return new SocketGuildChannel(guild.Discord, model.Id, guild);
```
Comment (Chris-Johnston) [inline src/Discord.Net.WebSocket/Entities/Channels/SocketGuildChannel.cs:39]: Should we call SocketGuildChannel.Update before returning? That's what SocketTextChannel and SocketVoiceChannel both do.

```csharp
default:
// TODO: Proper implementation for channel categories
SocketGuildChannel ret = new SocketGuildChannel(guild.Discord, model.Id, guild);
ret.Update(state, model);
return ret;
```

In my example case, IGuildChannels returned by `IGuild#GetChannelsAsync()` at least have their names. [Screenshot before/after.](http://i.imgur.com/kKnSltG.png)

Same could probably be applied to `RestGuildChannel`.

Comment (AntiTcb) [reply src/Discord.Net.WebSocket/Entities/Channels/SocketGuildChannel.cs:39]: Doing so would make the assumption that whatever unknown channel type existed follows the Channel model pattern, to some degree, which, in reality probably could be safely assumed, but I'd err on the side of skepticism here.

Comment (Chris-Johnston) [reply src/Discord.Net.WebSocket/Entities/Channels/SocketGuildChannel.cs:39]: Aren't we already making that assumption by returning a `SocketGuildChannel` in the first place for this patch?

Comment (AntiTcb) [reply src/Discord.Net.WebSocket/Entities/Channels/SocketGuildChannel.cs:39]: `SocketGuildChannel.Create()` is only ever used internally in a context where you'd only ever expect guild channels.

Having said that, I should probably also change [RestChannel](https://github.com/RogueException/Discord.Net/blob/dev/src/Discord.Net.Rest/Entities/Channels/RestChannel.cs) to cover those bases as well.

**Pull request merged**